### PR TITLE
feat(jenkins): migrate various <component>-master jobs and deis-v2-e2e(-master/-pr) job into DSL

### LINKED ISSUE #134: Migrate deis/charts related jobs to Jenkins Job DSL
The logic around the following jobs in ci.deis.io need to be moved to job dsl files in this repo:
- https://ci.deis.io/job/logger-master/
- https://ci.deis.io/job/minio-master/
- https://ci.deis.io/job/postgres-master/
- https://ci.deis.io/job/registry-master/
- https://ci.deis.io/job/builder-master/
- https://ci.deis.io/job/router-master/
- https://ci.deis.io/job/workflow-master/
- https://ci.deis.io/job/workflow-e2e-master/

and the most important one:
- https://ci.deis.io/job/deis-v2-e2e/

### PR DESCRIPTION
**Pull request by vdice**
TODOs:
- [x] remove `workflow-e2e-pr.groovy` DSL from [workflow-e2e](github.com/deis/workflow-e2e)
- [x] remove 'clone' from job names
- [x] ~~inspect Jenkins environment in `getProjectChannel()`~~

Closes #134

### CODE REVIEW

File: jobs/deis-v2-e2e.groovy [hunk 1]
```diff
@@ -0,0 +1,104 @@
+evaluate(new File("./common.groovy"))
+
+def getProjectChannel () {
+  repos.each { Map repo ->
+    // TODO: ? How to inspect jenkins environment to perform check below ?
+    if ("${repo.name.toUpperCase().replaceAll('-', '_')}_SHA" != '') {
```
Comment (vdice) [inline jobs/deis-v2-e2e.groovy]: @sgoings ^^ this not quite right.  Possible to inspect Jenkins env for particular value?

**Comment by vdice** [comment]
Please see current iterations describe in latest commit can be seen in https://ci.deis.io/job/deis-v2-e2e-pr-clone/configure and https://ci.deis.io/job/deis-v2-e2e-master-clone/configure and point out anything odd/missing...

File: jobs/common.groovy [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  [name: 'builder', slackChannel: builderChannel],
+  [name: 'dockerbuilder', slackChannel: builderChannel],
+  [name: 'logger', slackChannel: 'logger'],
+  [name: 'minio', slackChannel: 'object-store'],
+  [name: 'postgres', slackChannel: 'postgres'],
+  [name: 'registry', slackChannel: 'registry'],
+  [name: 'router', slackChannel: 'router'],
+  [name: 'slugbuilder', slackChannel: builderChannel],
+  [name: 'slugrunner', slackChannel: builderChannel],
+  [name: 'workflow', slackChannel: 'controller'],
```
Comment (helgi) [inline jobs/common.groovy]: time to change this to `controller`

File: jobs/deis-v2-e2e.groovy (hunk 1 above)
Comment (sgoings) [inline jobs/deis-v2-e2e.groovy]: My suggestion: just put the notifications in #testing or #engineering for now

**Comment by jchauncey** [comment]
This is awesome much :+1:

**Pull request merged**